btk/draw_blends.py

```diff
@@ -631,13 +635,14 @@ def render_mini_batch(self, blend_list, psf, wcs, survey, seed):
     def render_single(self, entry, filt, psf, survey, extra_data):
         """Returns the Galsim Image of an isolated galaxy."""
         base_image = extra_data
-        base_image = galsim.Convolve(base_image, psf)
+        gal_conv = galsim.Convolve(base_image, psf)
         gal_flux = get_flux(entry["ref_mag"], filt, survey)
-        base_image = base_image.withFlux(gal_flux)
-        base_image = base_image.shift(entry["ra"], entry["dec"])
-
+        gal_conv = gal_conv.withFlux(gal_flux)
+        gal_conv = gal_conv.shift(entry["ra"], entry["dec"])
+        if "g1" in entry.keys():
```

Comment:
this `if` statement can be avoided too if the default is 0.0 for shear everywhere in the code